### PR TITLE
Allow bool as an input type to built-in convert function

### PR DESCRIPTION
**Pull request by jakerockland**
### - What I did

Proposed solution to: #1064

### - How I did it

Updated `vyper/types/convert.py` to support boolean inputs.

### - How to verify it

I've added corresponding tests for the output types that can be convereted to from boolean values, so they can be run with:

`make test`

### - Description for the changelog

Updated built-in convert function to allow bool as an input type

### - Cute Animal Picture

image

### CODE REVIEW

File: tests/parser/types/test_bytes.py [hunk 1]
```diff
@@ -4,6 +4,19 @@
...
+    exp_code = """
+@public
+def testConvertBytes32(flag: bool) -> bytes32:
+    flagBytes: bytes32 = convert(flag, bytes32)
+    return flagBytes
+    """
+
+    c = get_contract_with_gas_estimation(exp_code)
+    assert c.testConvertBytes32(False) == b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+    assert c.testConvertBytes32(True) == b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01"
```
Comment (jacqueswww) [inline tests/parser/types/test_bytes.py]: I think just slicing `c.testConvertBytes32(True)[-1] == b'x01'` here would be more readable. If we want make sure about the 32 byte padding perhaps just an assert len() == 32 as well.

**Comment by jakerockland** [comment]
@jacqueswww Yeah that's a great point. I included the entire 32 bytes to make sure the padding was as expected, but I think making that change for readability and adding the second assert is the way to go. Just pushed a commit.

**Comment by jakerockland** [comment]
Fixes #1064

**Comment by jakerockland** [comment]
Added one last commit to fix style guideline issue that was throwing Travis CI off.

**Comment by fubuloubu** [comment]
@jakerockland, you're a beast!

**Comment by jakerockland** [comment]
Realized that there are already test files for conversions in the `tests/parser/functions` directory. I think some refactoring is still due per #1084, but think it would be better to refactor as I did in recent commits as opposed to all into one file.

**Pull request merged**